File: Python/Arrays&Hashing/TopKFrequentElements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        # Solution 1
        # Bucket Sort
        count = {}
        freq =[[] for i in range(len(nums)+1)]
        for n in nums:
            count[n] = 1 + count.get(n,0)
        for n, c in count.items():
            freq[c].append(n)
        res = []
        for i in range(len(freq)-1, 0, -1):
            for n in freq[i]:
                res.append(n)
                if len(res)==k:
                    return res

File: Go/Arrays&Hashing/TopKFrequentElements.go
func topKFrequent(nums []int, k int) (res []int) {
	countMap := map[int]int{}
	freq := make([][]int, len(nums)+1)

	for _, num := range nums {
		if count, ok := countMap[num]; ok {
			countMap[num] = count + 1
		} else {
			countMap[num] = 1
		}
	}

	for num, count := range countMap {
		freq[count] = append(freq[count], num)
	}

	for i := len(freq) - 1; i > 0; i-- {
		res = append(res, freq[i]...)
		if len(res) == k {
			return
		}
	}
	return
}
